mula/scheduler/queues/pq.py

```diff
@@ -139,24 +139,21 @@ def push(self, p_item: models.PrioritizedItem) -> Optional[models.PrioritizedIte
             # and we might need to update that.
             item_on_queue = self.get_p_item_by_identifier(p_item)
 
-            item_changed = (
-                False
-                if not item_on_queue or p_item.data == item_on_queue.data
-                else True  # FIXM: checking json/dicts here
+            item_changed = not (
+                not item_on_queue or p_item.data == item_on_queue.data
+            )  # FIXM: checking json/dicts here
+
+            priority_changed = not (not item_on_queue or p_item.priority == item_on_queue.priority)
+
+            allowed = any(
+                (
+                    item_on_queue and self.allow_replace,
+                    self.allow_updates and item_changed and item_on_queue,
+                    self.allow_priority_updates and priority_changed and item_on_queue,
+                    not item_on_queue,
+                )
             )
 
-            priority_changed = False if not item_on_queue or p_item.priority == item_on_queue.priority else True
-
-            allowed = False
-            if item_on_queue and self.allow_replace:
```

Comment:
Im not sure we should pack this code further. There's a reason some of this is written out like this.

Reply:
I think the linter in general did actually do a good job here. We could ignore specific statements for the linter, but in general, the SIM option improves and simplifies such expressions.

Rework: In this case, I managed to improve these expressions by applying De Morgan's law and negating the `==` operators which should be a bit easier to read now?